### PR TITLE
Fix bug in ARTIK board ERASE_ALL option

### PR DESCRIPTION
**Pull request by btheosam**
ARTIK05x download script doesn't erase parts of protected region
but it says successfully erased. This change would really erase all
the regions in FLASH. Please beware the after this change, ERASE_ALL
would erase, boot loaders, firmware, OS and applicaiton. Hence without
re-flashing appropriate binary, you won't even see serial logs.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [8d3fc598bdfe5024fac9e650461b5dff148d8dd3] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [8d3fc598bdfe5024fac9e650461b5dff148d8dd3] - Code Rule Check (C++) OK.

File: build/configs/artik05x/artik05x_download.sh [hunk 1]
```diff
@@ -188,7 +188,7 @@ erase()
 
 	# Download all binaries using openocd script
 	pushd ${OPENOCD_DIR_PATH} > /dev/null
-	${OPENOCD} -f ${CFG_FILE} -s ${SCRIPTS_PATH} -c "flash_erase_part ${PART_NAME};	exit" || exit 1
+	${OPENOCD} -f ${CFG_FILE} -s ${SCRIPTS_PATH} -c "flash_protect off; flash_erase_part ${PART_NAME};	exit" || exit 1
```
Comment (manoharahk) [inline build/configs/artik05x/artik05x_download.sh]: Do we require that extra "TAB" before exit?. I think one space is good

Comment (btheosam) [reply build/configs/artik05x/artik05x_download.sh]: Not necessary. Removed.

**Comment by seinfra** [comment]
Target : [132ccf28775c1394e2f89476c245c5503c9788b6] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [132ccf28775c1394e2f89476c245c5503c9788b6] - Code Rule Check (C++) OK.

File: build/configs/artik05x/artik05x_download.sh [hunk 2]
```diff
@@ -148,7 +148,7 @@ download()
 {
         parts_default=`grep -A 2 'config ARTIK05X_FLASH_PART_NAME' ${BOARD_KCONFIG} | sed -n 's/\tdefault "\(.*\)".*/\1/p'`
         parts2=${CONFIG_ARTIK05X_FLASH_PART_NAME:=${parts_default}}
-        parts=`echo $parts2 | sed "s/,/ /g"`
+        parts=${parts2//,/ }
```
Comment (sunghan-chang) [inline build/configs/artik05x/artik05x_download.sh]: Please don't mix a way of indentation. Basically this file has a tab as an indentation.

File: build/configs/artik05x/artik05x_download.sh [hunk 3]
```diff
@@ -188,7 +188,7 @@ erase()
 
 	# Download all binaries using openocd script
 	pushd ${OPENOCD_DIR_PATH} > /dev/null
-	${OPENOCD} -f ${CFG_FILE} -s ${SCRIPTS_PATH} -c "flash_erase_part ${PART_NAME};	exit" || exit 1
+	${OPENOCD} -f ${CFG_FILE} -s ${SCRIPTS_PATH} -c "flash_protect off; flash_erase_part ${PART_NAME}; exit" || exit 1
```
Comment (sunghan-chang) [inline build/configs/artik05x/artik05x_download.sh]: Do you test this with partial region like os?
When we erase os region, new command is not necessary. It needs only for bootloader.
Doesn't it cause any error?

Comment (btheosam) [reply build/configs/artik05x/artik05x_download.sh]: NO, it wont cause any error. Just that flash protection would be turned off.

Comment (sunghan-chang) [reply build/configs/artik05x/artik05x_download.sh]: With all option, you're right. But with others, you're wrong.
When I erase userfs for example, after executing it, flash will be protected off even I didn't tough bootloader.
It's definitely wrong.
`Protect off` should be executed when we treat bootloader regions.

Comment (btheosam) [reply build/configs/artik05x/artik05x_download.sh]: Will rework soon.

**Comment by seinfra** [comment]
Target : [548fee3260b5b9a6e3314c4ff68ae0c520dc2364] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [548fee3260b5b9a6e3314c4ff68ae0c520dc2364] - Code Rule Check (C++) OK.

File: build/configs/artik05x/artik05x_download.sh (hunk 2 above)
Comment (btheosam) [reply build/configs/artik05x/artik05x_download.sh]: fixed

File: build/configs/artik05x/artik05x_download.sh [hunk 4]
```diff
@@ -167,28 +167,30 @@ download()
...
 {
+	# Generate Partition Map
+	${SCRIPTS_PATH}/partition_gen.sh
+
+	pushd ${OPENOCD_DIR_PATH} > /dev/null
+
 	case $optarg in
 		OTA|ota)
 			PART_NAME=ota
+			${OPENOCD} -f ${CFG_FILE} -s ${SCRIPTS_PATH} -c "flash_erase_part ${PART_NAME}; exit" || exit 1
```
Comment (sunghan-chang) [inline build/configs/artik05x/artik05x_download.sh]: Repeatation looks not good.
For example, how about below?
```
  case ..
    OTA|ota)
      PART_NAME=ota
      ;;
    USERFS|userfs)
      PART_NAME=user
      ;;
    ALL|all)
      PART_NAME=all
      NEED_PROTECTCMD=y
      ;;
    *)
      ....
  esac

  if [ "${NEED_PROTECTCMD}" == "y" ]; then
    PROTECT_OFFCMD="flash_protect off"
    PROTECT_ONCMD="flash_protect on"
  fi

 # Generate ...
 ${SCRIPTS_PATH}/....

  # Download all ...
  pushd ...
  ${OPENOCD} -f ${CFG_FILE} -s ${SCRIPTS_PATH} -c "${PROTECT_OFFCMD}; flash_erase_part ${PART_NAME}; ${PROTECT_ONCMD}; exit" || exit 1
  popd ...
```

Comment (btheosam) [reply build/configs/artik05x/artik05x_download.sh]: This gives the flexibility to modify PATH, CMDs and attributes(present and future) to each partition.
But if reviewer wants to tie and bound all attributes(present and future) , erase functionalities, etc to one command then this comment can be incorporated.  (As a matter of fact, its all about flexibility, readability and maintainability vs repetitions). Anything is okay.

**Comment by seinfra** [comment]
Target : [e5c458141abda0468a1a0d0840b1243422b07933] -

**Comment by seinfra** [comment]
Target : [6356af77151e1987b6d893a2af87f5476729ef53] -

**Comment by seinfra** [comment]
Target : [9029418857487bf0100e21c1b1801c2cb172f1c9] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [9029418857487bf0100e21c1b1801c2cb172f1c9] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [0c61a1cab76ae7fbf699eeb737fce4d5dee65509] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [0c61a1cab76ae7fbf699eeb737fce4d5dee65509] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [c194f29e7ca413e398ae72b4fe13817e843b0f71] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [c194f29e7ca413e398ae72b4fe13817e843b0f71] - Code Rule Check (C++) OK.

**Comment by btheosam** [comment]
Had sever merge conflicts with other commits and hence the branch gets corrupted and hence got removed.

**Comment by seinfra** [comment]
Target : [59bb4986c1e4aff7fd87ba3d9a0e722016c012d2] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [59bb4986c1e4aff7fd87ba3d9a0e722016c012d2] - Code Rule Check Result:
apps/examples/testcase/le_tc/kernel/tc_libc_math.c:2623: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:WxV)
framework/src/media/Decoder.h:39: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/Decoder.h:39: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/Decoder.h:39: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/Decoder.h:39: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/utils/MediaUtils.h:44: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/utils/MediaUtils.h:44: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/utils/MediaUtils.h:44: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/utils/MediaUtils.h:51: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/utils/MediaUtils.h:51: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/utils/MediaUtils.h:51: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/utils/MediaUtils.h:59: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/utils/MediaUtils.h:59: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/utils/MediaUtils.h:67: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/utils/MediaUtils.h:67: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
os/board/common/common.h:26: ERROR: [IDT_M_TAB] please, no spaces at the start of a line

**Pull request closed without merging**